src/views/network/view_network.cpp

```diff
@@ -81,22 +81,43 @@ namespace big
 
 	void render_session_switcher()
 	{
+		if (g_pointers->m_gta.m_region_code == nullptr)
+			return;
+
+		static int selected_region_index     = -1;
+		static bool region_updated = false;
+
+        std::string region_str = (selected_region_index == -1) ? "SESSION_SELECT_COMBO"_T.data() : regions[*g_pointers->m_gta.m_region_code].name;
+
 		ImGui::BeginGroup();
-		components::sub_title("SESSION_SWITCHER"_T);
+		components::sub_title("SESSION_SELECT"_T);
 		if (ImGui::BeginListBox("###session_switch", get_listbox_dimensions()))
 		{
-			if (ImGui::BeginCombo("##regionswitcher", "REGIONS"_T.data()))
+			if (ImGui::BeginCombo("##regionswitcher", region_str.c_str()))
 			{
 				for (const auto& region_type : regions)
 				{
-					components::selectable(region_type.name, *g_pointers->m_gta.m_region_code == region_type.id, [&region_type] {
-						*g_pointers->m_gta.m_region_code = region_type.id;
-					});
+                    components::selectable(region_type.name, *g_pointers->m_gta.m_region_code == region_type.id, [&region_type] 
+                    {
+                        *g_pointers->m_gta.m_region_code = region_type.id;
+                        region_updated = true;
+                    });
```

Comment:
clang-format hasn't ran on this.

Reply:
Ugh. I need to get better about this. Updated.